### PR TITLE
Release notes for 3.5.2

### PR DESCRIPTION
**Pull request by lorenrh**
Signed-off-by: Lorena Rangel <lorena.rangel@docker.com>

### CODE REVIEW

File: docker-for-mac/release-notes.md [hunk 1]
```diff
@@ -25,6 +25,32 @@ This page contains information about the new features, improvements, known issue
...
+- Fix spurious traces on iptables updates.
+- Fix slowness when adding multiple ports forwarding option.
+
+## Docker Desktop 3.5.1
+2021-06-25
+
+[Download for Mac with Intel chip](https://desktop.docker.com/mac/stable/amd64/66090/Docker.dmg)
+[Download for Mac with Apple chip](https://desktop.docker.com/mac/stable/arm64/66090/Docker.dmg)
+
+### New
```
Comment (StefanScherer) [inline docker-for-mac/release-notes.md:48]: We should move the New section to the top, and remove the duplicate Compose v2 beta section.

File: docker-for-mac/release-notes.md [hunk 2]
```diff
@@ -25,6 +25,32 @@ This page contains information about the new features, improvements, known issue
...
 
+### Upgrades
+
+- [Compose V2 beta](https://github.com/docker/compose-cli/releases/tag/v2.0.0-beta.5)
+  - `compose run` and `compose exec` use separate streams for stdout and stderr
+  - `compose run` and `compose exec` support detach keys
+  - fixed `--force` and `--volumes` flags on compose rm command
+  - fixed network's IPAM configuration. Service can define a fixed IP
+
```
Comment (gtardif) [inline docker-for-mac/release-notes.md:35]: - Dev Environments
  - Support VSCode insiders. See https://github.com/docker/dev-environments/issues/3
  - Allow to specify a branche when cloning. See https://github.com/docker/dev-environments/issues/11

File: docker-for-mac/release-notes.md [hunk 3]
```diff
@@ -25,6 +25,32 @@ This page contains information about the new features, improvements, known issue
...
+### Bug fixes and minor changes
+
+- Fix connection leak when using `virtualization.framework` and not using `vpnkit`.
+- Fix spurious traces on iptables updates.
+- Fix slowness when adding multiple ports forwarding option.
+
```
Comment (gtardif) [inline docker-for-mac/release-notes.md:41]: - Dev Environments: Fix blank screen in some create/remove scenarios. Fixes https://github.com/docker/dev-environments/issues/4
- Dev Environments: Fix error handling in removing an environment. Fixes https://github.com/docker/dev-environments/issues/8
- Dev Environments: Disable Start/Stop/Share buttons while an environment is being created or removed

**Review by gtardif — COMMENTED** [review]
Some additional things for Dev Environments

File: docker-for-mac/release-notes.md (hunk 1 above)
Comment (lorenrh) [reply docker-for-mac/release-notes.md:48]: oh right! I didn't copy everything! Thanks!

File: docker-for-mac/release-notes.md [hunk 4]
```diff
@@ -13,8 +13,8 @@ This page contains information about the new features, improvements, known issue
 
 {% include eula.md %}
 
-## Docker Desktop 3.5.1
-2021-06-25
+## Docker Desktop 3.5.2
```
Comment (mat007) [inline docker-for-mac/release-notes.md:16]: Is this replacing 3.5.1 with 3.5.2?

Comment (lorenrh) [reply docker-for-mac/release-notes.md:16]: Not really, the 3.5.1 release notes are bellow.

File: docker-for-mac/release-notes.md [hunk 5]
```diff
@@ -25,6 +25,32 @@ This page contains information about the new features, improvements, known issue
...
 
+### Upgrades
+
+- [Compose V2 beta](https://github.com/docker/compose-cli/releases/tag/v2.0.0-beta.5)
+  - `compose run` and `compose exec` use separate streams for stdout and stderr
+  - `compose run` and `compose exec` support detach keys
+  - fixed `--force` and `--volumes` flags on compose rm command
+  - fixed network's IPAM configuration. Service can define a fixed IP
```
Comment (mat007) [inline docker-for-mac/release-notes.md:34]: Should we link to the `compose-cli` issues?

File: docker-for-mac/release-notes.md (hunk 4 above)
Comment (mat007) [reply docker-for-mac/release-notes.md:16]: Ah it’s the diff getting confused actually…

File: docker-for-mac/release-notes.md (hunk 2 above)
Comment (lorenrh) [reply docker-for-mac/release-notes.md:38]: Done!

File: docker-for-mac/release-notes.md (hunk 3 above)
Comment (lorenrh) [reply docker-for-mac/release-notes.md:47]: Added in both mac and win!

File: docker-for-mac/release-notes.md [hunk 6]
```diff
@@ -23,6 +23,38 @@ This page contains information about the new features, improvements, known issue
 
 **Dev Environments Preview**: Dev Environments enable you to seamlessly collaborate with your team members without moving between Git branches to get your code onto your team members' machines. When using Dev Environments, you can share your in-progress work with your team members in just one click, and without having to deal with any merge conflicts. For more information and for instructions on how to use Dev Environments, see [Development Environments Preview](../desktop/dev-environments.md).
 
+
+### Upgrades
+
+- [Compose V2 beta 6](https://github.com/docker/compose-cli/releases/tag/v2.0.0-beta.6)
+  - `compose run` and `compose exec` use separate streams for stdout and stderr. See [docker/compose-cli#1873](https://github.com/docker/compose-cli/issues/1873).
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:30]:
```suggestion
  - `compose run` and `compose exec` commands now use separate streams for stdout and stderr. See [docker/compose-cli#1873](https://github.com/docker/compose-cli/issues/1873).
```

File: docker-for-mac/release-notes.md [hunk 7]
```diff
@@ -23,6 +23,38 @@ This page contains information about the new features, improvements, known issue
...
+  - `compose run` and `compose exec` support detach keys. Fixes [docker/compose-cli#1709](https://github.com/docker/compose-cli/issues/1709).
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:31]:
```suggestion
  - `compose run` and `compose exec` commands support detach keys. Fixes [docker/compose-cli#1709](https://github.com/docker/compose-cli/issues/1709).
```

File: docker-for-mac/release-notes.md [hunk 8]
```diff
@@ -23,6 +23,38 @@ This page contains information about the new features, improvements, known issue
...
+  - fixed `--force` and `--volumes` flags on compose rm command. See [docker/compose-cli#1844](https://github.com/docker/compose-cli/issues/1844).
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:32]:
```suggestion
  - Fixed `--force` and `--volumes` flags on `compose rm` command. See [docker/compose-cli#1844](https://github.com/docker/compose-cli/issues/1844).
```

File: docker-for-mac/release-notes.md [hunk 9]
```diff
@@ -23,6 +23,38 @@ This page contains information about the new features, improvements, known issue
...
+  - fixed network's IPAM configuration. Service can define a fixed IP. Fixes for [docker/compose-cli#1678](https://github.com/docker/compose-cli/issues/1678) and [docker/compose-cli#1816](https://github.com/docker/compose-cli/issues/1816)
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:33]:
```suggestion
  - Fixed network's IPAM configuration. Service can define a fixed IP. Fixes [docker/compose-cli#1678](https://github.com/docker/compose-cli/issues/1678) and [docker/compose-cli#1816](https://github.com/docker/compose-cli/issues/1816)
```

File: docker-for-mac/release-notes.md [hunk 10]
```diff
@@ -23,6 +23,38 @@ This page contains information about the new features, improvements, known issue
...
+
+- Dev Environments
+  - Support VSCode insiders. See [dev-environments#3](https://github.com/docker/dev-environments/issues/3)
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:36]:
```suggestion
  - Support for VS Code Insiders. See [dev-environments#3](https://github.com/docker/dev-environments/issues/3)
```

File: docker-for-mac/release-notes.md [hunk 11]
```diff
@@ -23,6 +23,38 @@ This page contains information about the new features, improvements, known issue
...
+  - Allow to specify a branch when cloning. See [dev-environments#11](https://github.com/docker/dev-environments/issues/11)
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:37]:
```suggestion
  - Allow users to specify a branch when cloning a project. See [dev-environments#11](https://github.com/docker/dev-environments/issues/11)
```

File: docker-for-mac/release-notes.md [hunk 12]
```diff
@@ -23,6 +23,38 @@ This page contains information about the new features, improvements, known issue
...
+
+### Bug fixes and minor changes
+
+- Dev Environments: Fix blank screen in some create/remove scenarios. Fixes [dev-environments#4](https://github.com/docker/dev-environments/issues/4)
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:41]:
```suggestion
- Dev Environments: Fixed a blank screen in some create and remove scenarios. Fixes [dev-environments#4](https://github.com/docker/dev-environments/issues/4)
```

File: docker-for-mac/release-notes.md [hunk 13]
```diff
@@ -23,6 +23,38 @@ This page contains information about the new features, improvements, known issue
...
+- Dev Environments: Fix error handling in removing an environment. Fixes [dev-environments#8](https://github.com/docker/dev-environments/issues/8)
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:42]:
```suggestion
- Dev Environments: Fixed error handling when removing an environment. Fixes [dev-environments#8](https://github.com/docker/dev-environments/issues/8)
```

File: docker-for-mac/release-notes.md [hunk 14]
```diff
@@ -23,6 +23,38 @@ This page contains information about the new features, improvements, known issue
...
+- Dev Environments: Disable Start/Stop/Share buttons while an environment is being created or removed
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:43]:
```suggestion
- Dev Environments: The **Start**, **Stop**, and **Share** buttons are disabled while an environment is being created or removed.
```

File: docker-for-mac/release-notes.md [hunk 15]
```diff
@@ -23,6 +23,38 @@ This page contains information about the new features, improvements, known issue
...
+- Fix connection leak when using `virtualization.framework` and not using `vpnkit`.
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:44]:
```suggestion
- Fixed a connection leak when using `virtualization.framework` and not using `vpnkit`.
```

File: docker-for-mac/release-notes.md [hunk 16]
```diff
@@ -23,6 +23,38 @@ This page contains information about the new features, improvements, known issue
...
+- Fix spurious traces on iptables updates.
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:45]:
```suggestion
- Fixed spurious traces on iptables updates.
```

File: docker-for-mac/release-notes.md [hunk 17]
```diff
@@ -23,6 +23,38 @@ This page contains information about the new features, improvements, known issue
...
+- Fix slowness when adding multiple ports forwarding option.
```
Comment (usha-mandya) [inline docker-for-mac/release-notes.md:46]:
```suggestion
- Fixed a delay when adding a multiple port forwarding option.
```

File: docker-for-windows/release-notes.md [hunk 18]
```diff
@@ -13,10 +13,41 @@ This page contains information about the new features, improvements, known issue
...
+- Dev Environments
+  - Support VS Code Insiders. See [dev-environments#3](https://github.com/docker/dev-environments/issues/3)
+  - Allow users to specify a branch when cloning a project. See [dev-environments#11](https://github.com/docker/dev-environments/issues/11)
+
+### Bug fixes and minor changes
+
+- Dev Environments: Fixed a blank screen in some create and remove scenarios. Fixes [dev-environments#4](https://github.com/docker/dev-environments/issues/4)
+- Dev Environments: Fixed error handling when removing an environment. Fixes [dev-environments#8](https://github.com/docker/dev-environments/issues/8)
+- Dev Environments: The **Start**, **Stop**, and **Share** buttons are disabled while an environment is being created or removed.
+- Do not automatically switch CLI contexts on application start or when switching between Windows and Linux containers. Fixes [docker/for-mac#5787](https://github.com/docker/for-mac/issues/5787).
```
Comment (StefanScherer) [inline docker-for-windows/release-notes.md:43]: Maybe we should also link https://github.com/docker/for-win/issues/11530 and close with 3.5.2.

**Review by usha-mandya — APPROVED** [review]
LGTM

**Pull request merged**